ui/StatusQ/src/wallet/managetokenscontroller.cpp

```diff
@@ -476,6 +525,22 @@ void ManageTokensController::setArrangeByCommunity(bool newArrangeByCommunity)
     emit arrangeByCommunityChanged();
 }
 
+bool ManageTokensController::arrangeByCollection() const
+{
+    return m_arrangeByCollection;
+}
+
+void ManageTokensController::setArrangeByCollection(bool newArrangeByCollection)
+{
+    if(m_arrangeByCollection == newArrangeByCollection) return;
+    m_arrangeByCollection = newArrangeByCollection;
+    if (!m_arrangeByCollection)
+        m_regularTokensModel->applySort();
+    else
+        rebuildRegularTokenGroupsModel();
```

Comment:
You don't need to "rebuild" here, just keep the provious individual items order as it was.

Reply:
I'll remove the opposite branch; that is we throw away any changes done in the "grouped" mode